include/UIMenu.h

```diff
@@ -36,99 +37,104 @@
 class UIMenu;
 class Page;
 
-//typedef void(*MenuCallback)(UIMenu*);
+// typedef void(*MenuCallback)(UIMenu*);
 typedef std::function<void(UIMenu*)> MenuCallback;
 typedef std::function<void(UIMenu*, void*)> ParameterizedMenuCallback;
 typedef std::function<void(UIMenu*, int)> IParameterizedMenuCallback;
+typedef std::function<void(UIMenu*, void*, int)> PIParameterizedMenuCallback;
 
 enum AutocleanMethod {
-  AUTOCLEAN_OFF,
-  AUTOCLEAN_FREE,
-  AUTOCLEAN_DELETE,
+    AUTOCLEAN_OFF,
+    AUTOCLEAN_FREE,
+    AUTOCLEAN_DELETE,
 };
 
 typedef struct UIMenuItem {
-  char text[21];
-  MenuCallback cb;
-  ParameterizedMenuCallback pcb;
-  IParameterizedMenuCallback ipcb;
-  void *arg = nullptr;
-  int iarg = NULL;
-  UIMenuItem *next = nullptr;
-  UIMenuItem *prev = nullptr;
+    char text[21];
+    MenuCallback cb;
+    ParameterizedMenuCallback pcb;
+    IParameterizedMenuCallback ipcb;
+    PIParameterizedMenuCallback pipcb;
```

Comment:
All callbacks will become PI callbacks, and this will be the new standard going forward.